### PR TITLE
Added support for multiple database connections

### LINKED ISSUE #4: DB transactions with multiple connections
Strategies that utilize database transactions do not work correctly with multiple database connections configured. It always uses the default connection:

[DatabaseTransactionHelpers.php](https://github.com/knuckleswtf/scribe/blob/master/src/Extracting/DatabaseTransactionHelpers.php)
```
app('db')->beginTransaction();
```

I started adding support for multiple connections in [my fork](https://github.com/hendrikbl/scribe/tree/multi-connection). It simply gets the connection from the model type and passes it to the `startDbTransaction` and `endDbTransaction` function.

I'm not sure though how to do it in [ResponseCalls.php](https://github.com/knuckleswtf/scribe/blob/master/src/Extracting/Strategies/Responses/ResponseCalls.php) since I don't know any way to get the utilized connection from the route itself. One way could be simply iterating over all connections and starting and ending transactions for all connections. But if there is a better method I'd prefer that :D

Let me know what you think and if i missed anything.

### PR DESCRIPTION
**Pull request by hendrikbl**
Closes #4 

For ApiResource and Transformer strategies the model types connection is used to start the transaction. For the ResponseCall strategy a transaction for every database connection is started since we don't know which one is used for a specific route.

### CODE REVIEW

File: src/Extracting/DatabaseTransactionHelpers.php [hunk 1]
```diff
@@ -7,23 +7,27 @@
 trait DatabaseTransactionHelpers
 {
     /**
+     * @param string $connection
+     *
      * @return void
      */
-    private function startDbTransaction()
+    private function startDbTransaction(String $connection = null)
```
Comment (shalvah) [inline src/Extracting/DatabaseTransactionHelpers.php:14]: Canonically, in PHP, it's `string`, not `String`. Let's keep it uniform.

File: src/Extracting/Strategies/Responses/ResponseCalls.php [hunk 2]
```diff
@@ -90,7 +90,11 @@ public function makeResponseCall(Route $route, array $context, array $rulesToApp
      */
     private function configureEnvironment(array $rulesToApply)
     {
-        $this->startDbTransaction();
+        // Start transactions for all connections since we don't know wich one is used
+        $connections = array_keys(config('database.connections'));
```
Comment (shalvah) [inline src/Extracting/Strategies/Responses/ResponseCalls.php:94]: Can we rather move this whole logic into the startDbTransactions method? Same for the endDbTransactions.The configureEnvironment method should just be a high-level list of steps.

Also, don't forget to wrap each individual call to app('db')->... in a try/catch.

**Comment by shalvah** [comment]
Another issue iterating over connections would solve: models that don't extend Eloquent.

File: src/Extracting/DatabaseTransactionHelpers.php (hunk 1 above)
Comment (hendrikbl) [reply src/Extracting/DatabaseTransactionHelpers.php:14]: If we take the other approach for everything, we don't need this anyways :)

**Comment by hendrikbl** [comment]
Let me know if there is something left to do :)

File: src/Extracting/Strategies/Responses/ResponseCalls.php (hunk 2 above)
Comment (hendrikbl) [reply src/Extracting/Strategies/Responses/ResponseCalls.php]: Done!

File: tests/Fixtures/TestModel.php [hunk 3]
```diff
@@ -2,12 +2,14 @@
...
 
+use Illuminate\Database\Eloquent\Model;
+
 /**
  * A demo test model.
  *
  * @author Tobias van Beek <t.vanbeek@tjvb.nl>
  */
-class TestModel
+class TestModel extends Model
```
Comment (shalvah) [inline tests/Fixtures/TestModel.php:12]:
```suggestion
class TestModel
```

**Pull request merged**